Build is failing because of incorrect definition of long description

Hi,

we still have the issue that some long descriptions will fail the complete build. Please see the following output
```
10:41:28 [resources] $ /home/ubuntu/.ant/bin/ant -file build.xml -DghprbActualCommitAuthorEmail=ivan.senic@novatec-gmbh.de -DghprbTriggerAuthorLogin= -DghprbGhRepository=inspectIT/inspectIT -DghprbTargetBranch=master -DghprbPullId=9 -DghprbCredentialsId=f9564afd-e9e5-4935-8957-e44dace783d8 -DghprbAuthorRepoGitUrl=https://github.com/ivansenic/inspectIT.git -DghprbTriggerAuthor= -DghprbPullAuthorEmail= -DghprbPullLink=https://github.com/inspectIT/inspectIT/pull/9 -Dsha1=origin/pr/9/merge -DghprbPullAuthorLoginMention=@ivansenic "-DghprbPullDescription=GitHub pull request #9 of commit 75525ce2ef82cd2d493d2f0cb96c19a2b13a92b1, no merge conflicts." -DghprbActualCommit=75525ce2ef82cd2d493d2f0cb96c19a2b13a92b1 -DghprbPullAuthorLogin=ivansenic -DghprbSourceBranch=INSPECTIT-1938 "-DghprbActualCommitAuthor=Ivan Senic" -DghprbTriggerAuthorLoginMention= '-DghprbPullLongDescription=~~Note that commons-net is already included in the THIRDPARTYLICENSE.txt~~
\n
\n_As we moved to github API Commons Net are removed and I added Gson as new dependency._
\n
\n<!-- Reviewable:start -->
\n[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/inspectit/inspectit/9)
\n<!-- Reviewable:end -->
\n' -DghprbTriggerAuthorEmail= -DGIT_BRANCH=INSPECTIT-1938 "-DghprbPullTitle=INSPECTIT-1938: (Auto-)Check for new version available in UI" -DghprbCommentBody=null clean
10:41:28 Buildfile: /home/ubuntu/jenkins/workspace/inspectIT - Pull Request Check/Commons/resources/build.xml
10:41:29 
10:41:29 BUILD FAILED
10:41:29 Target "on" does not exist in the project "inspectIT - Commons". 
10:41:29 
10:41:29 Total time: 0 seconds
10:41:29 Build step 'Invoke Ant' marked build as failure
```

Is there somehow an easy fix for this? Because of this problem, we have to revert to a quite old version of the plugin. Can we maybe just ignore that long description completely so that it will not even appear in the execution?

Cheers,
Patrice

--- a/src/main/java/org/jenkinsci/plugins/ghprb/GhprbTrigger.java
+++ b/src/main/java/org/jenkinsci/plugins/ghprb/GhprbTrigger.java
@@ -287,12 +287,12 @@ public class GhprbTrigger extends GhprbTriggerBackwardsCompatible {
         values.add(new StringParameterValue("ghprbPullAuthorLogin", String.valueOf(cause.getPullRequestAuthor().getLogin())));
         values.add(new StringParameterValue("ghprbPullAuthorLoginMention", "@" + cause.getPullRequestAuthor().getLogin()));
         
-        values.add(new StringParameterValue("ghprbPullDescription", String.valueOf(cause.getShortDescription()).replace("\n", "\\n")));
+        values.add(new StringParameterValue("ghprbPullDescription", escapeText(String.valueOf(cause.getShortDescription()))));
         values.add(new StringParameterValue("ghprbPullTitle", String.valueOf(cause.getTitle())));
         values.add(new StringParameterValue("ghprbPullLink", String.valueOf(cause.getUrl())));
-        values.add(new StringParameterValue("ghprbPullLongDescription", String.valueOf(cause.getDescription()).replace("\n", "\\n")));
+        values.add(new StringParameterValue("ghprbPullLongDescription", escapeText(String.valueOf(cause.getDescription()))));
         
-        values.add(new StringParameterValue("ghprbCommentBody", String.valueOf(cause.getCommentBody()).replace("\n", "\\n")));
+        values.add(new StringParameterValue("ghprbCommentBody", escapeText(String.valueOf(cause.getCommentBody()))));
         
         values.add(new StringParameterValue("ghprbGhRepository", repo.getName()));
         values.add(new StringParameterValue("ghprbCredentialsId", getString(getGitHubApiAuth().getCredentialsId(), "")));
@@ -302,7 +302,10 @@ public class GhprbTrigger extends GhprbTriggerBackwardsCompatible {
         // one isn't there
         return this.job.scheduleBuild2(job.getQuietPeriod(), cause, new ParametersAction(values), findPreviousBuildForPullId(pullIdPv));
     }
-    
+
+    private String escapeText(String text) {
+        return text.replace("\n", "\\n").replace("\r", "\\r").replace("\"", "\\\"");
+    }
     
     public GhprbGitHubAuth getGitHubApiAuth() {
         if (gitHubAuthId == null) {

